libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -699,16 +699,56 @@ string YulUtilFunctions::storageArrayPopFunction(ArrayType const& _type)
 				let oldLen := <fetchLength>(array)
 				if iszero(oldLen) { invalid() }
 				let newLen := sub(oldLen, 1)
+				let slot, offset := <indexAccess>(array, newLen)
+				<setToZero>(slot, offset)
+				sstore(array, newLen)
+			})")
+			("functionName", functionName)
+			("fetchLength", arrayLengthFunction(_type))
+			("indexAccess", storageArrayIndexAccessFunction(_type))
+			("setToZero", storageSetToZeroFunction(*_type.baseType()))
+			.render();
+	});
+}
+
+string YulUtilFunctions::storageByteArrayPopFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	solAssert(_type.isByteArray(), "");
 
+	string functionName = "array_pop_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array) {
+				let oldLen := <fetchLength>(array)
+				if iszero(oldLen) { invalid() }
+				let newLen := sub(oldLen, 1)
 				let slot, offset := <indexAccess>(array, newLen)
 				<setToZero>(slot, offset)
```

Comment:
I think this can still be specialized further. This already sstores to `array` in case the length is less than 32 bytes.

I mean I'm a friend of modularity, but somehow I feel that in this case, it is better to make it visible that data and length share the same slot and that this function actually writes to that slot for short arrays.